src/views/MyProfile/Components/VictoryPromiseDisplay/index.js

```diff
@@ -0,0 +1,340 @@
+import Grid from '@material-ui/core/Grid';
+import React from 'react';
+import CardHeader from '@material-ui/core/CardHeader';
+import Tooltip from '@material-ui/core/Tooltip';
+import { gordonColors } from '../../../../theme';
+import { Polar } from 'react-chartjs-2';
+import victory from '../../../../services/victory';
+import './VictoryPromise.css';
+import Button from '@material-ui/core/Button';
+import './VictoryPromise.css';
+import { withStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+import Card from '@material-ui/core/Card';
+import CardContent from '@material-ui/core/CardContent';
+
+export default class VictoryPromiseDisplay extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      labels: [
+        'Christian Character',
+        'Intellectual Maturity',
+        'Lives of Service',
+        'Leadership Worldwide',
+      ],
+      datasets: [
+        {
+          data: [1, 1, 1, 1],
+          backgroundColor: [
+            gordonColors.neutral.lightGray,
+            gordonColors.neutral.lightGray,
+            gordonColors.neutral.lightGray,
+            gordonColors.neutral.lightGray,
+          ],
+        },
+      ],
+      options: '',
+      CC: 0,
+      IM: 0,
+      LS: 0,
+      LW: 0,
+      CCColor: gordonColors.neutral.lightGray,
+      IMColor: gordonColors.neutral.lightGray,
+      LSColor: gordonColors.neutral.lightGray,
+      LWColor: gordonColors.neutral.lightGray,
+      defaultVPMode: true,
+      CC_ON: false,
+      IM_ON: false,
+      LS_ON: false,
+      LW_ON: false,
+    };
+  }
+
+  changeMode = () => {
+    console.log(this.state.defaultVPMode);
+    if (this.state.defaultVPMode) {
+      this.setState({ defaultVPMode: false });
+    } else {
+      this.setState({ defaultVPMode: true });
+    }
+  };
+
+  componentWillMount() {
+    this.getVPScores();
+  }
+
+  async getVPScores() {
+    const scores = await victory.getVPScore();
+    //const CC = scores[0].TOTAL_VP_CC_SCORE;
+    const CC = scores[0].TOTAL_VP_CC_SCORE;
+    const IM = scores[0].TOTAL_VP_IM_SCORE;
+    const LS = scores[0].TOTAL_VP_LS_SCORE;
+    const LW = scores[0].TOTAL_VP_LW_SCORE;
+    this.setState({ CC, IM, LS, LW });
+
+    if (CC > 0) {
+      this.setState({ CCColor: gordonColors.secondary.red, CC_ON: true });
+    }
+    if (IM > 0) {
+      this.setState({ IMColor: gordonColors.secondary.green, IM_ON: true });
+    }
+    if (LS > 0) {
+      this.setState({ LSColor: gordonColors.secondary.yellow, LS_ON: true });
+    }
+    if (LW > 0) {
+      this.setState({ LWColor: gordonColors.primary.cyan, LW_ON: true });
+    }
+
+    this.setState({
+      options: {
```

Comment:
Can we not initialize these values in the constructor?

Reply:
what values? CC, IM, LS, LW? or options? 
We are getting these values from victory.getVPScore(), so I think it can't go in the constructor

Reply:
what values? CC, IM, LS, LW? or options? 
We are getting these values from victory.getVPScore(), so I think it can't go in the constructor